### ISSUE TITLE
`pm2 startup` does not take into account authbind nor `pm2` alias

### ISSUE THREAD

**Issue by niftylettuce** [Unitech/pm2#3165]
In the docs we suggest to users to add the following line to their non-root user's `~/.bashrc` file.

```sh
alias pm2='authbind --deep pm2'
```

However this ONLY works for when the user is directly logged in as the user.

Upon calling `pm2 save`, the following output occurs:

```sh
user@ubuntu:~$ pm2 save
[PM2] Saving current process list...
[PM2] Successfully saved in /home/user/.pm2/dump.pm2
```

This is OK actually, the problem is the next step:

```sh
user@ubuntu-2gb-nyc3-01:~$ pm2 startup
[PM2] Init System found: upstart
[PM2] You have to run this command as root. Execute the following command:
sudo env PATH=$PATH:/home/user/.nvm/versions/node/v8.3.0/bin /home/user/.config/yarn/global/node_modules/pm2/bin/pm2 startup upstart -u user --hp /home/user
```

Note that the `pm2` command here does not utilize `authbind`.

Therefore when you run that command as root, and then run `sudo reboot` to reboot the server, the ONLY way your apps will run again properly is if you manually log in as `user` and run:

```sh
authbind --deep pm2 update
```

I think that we should detect if the user is using authbind and then insert that in the copy/paste snippet.  Otherwise we have to manually modify it as such:

```sh
sudo env PATH=$PATH:/home/user/.nvm/versions/node/v8.3.0/bin /usr/bin/authbind --deep /home/user/.config/yarn/global/node_modules/pm2/bin/pm2 startup upstart -u user --hp /home/user
```

This issue is related to the comment here <https://github.com/Unitech/PM2/issues/1038#issuecomment-211459999>.

Perhaps allow the user to pass a flag `--authbind` to `pm2 startup` command?  And then take that into consideration here <https://github.com/Unitech/pm2/blob/43b6000171a28e09cc25467e1b46fcd7f7ac7a08/lib/API/Startup.js#L281>?

**Comment by niftylettuce** [comment]
In the init script I think it needs modified?

```sh
PM2=/home/user/.config/yarn/global/node_modules/pm2/bin/pm2
```

Needs to be:

```sh
PM2=/usr/bin/authbind --deep /home/user/.config/yarn/global/node_modules/pm2/bin/pm2
```

Otherwise we'd have to run as the user `authbind --deep pm2 update`?

**Comment by niftylettuce** [comment]
This seems to work OK - just rebooted server and it worked OK; so that is the solution!  So perhaps we just need a flag `--authbind` or `--with-authbind` added to `pm2 startup` to know that it needs to prepend this on the line mentioned in my previous comment.

**Comment by vmarchaud** [comment]
Indeed, we are aware of this problem, as said on slack : 
``` should pm2 startup register the service with authbind when available on the system ? ```

**Comment by niftylettuce** [comment]
Yeah it seems insane to me that this isn't the default already and nobody else has complained about this really 😦

**Comment by niftylettuce** [comment]
Can we get a fix in asap? @vmarchaud

**Comment by vmarchaud** [comment]
Don't have the time right now, feel free to make a PR
Related file is [right there](https://github.com/Unitech/pm2/blob/master/lib/API/Startup.js#L165), you have each init system handled separately inside a switch.

**Comment by damianobarbati** [comment]
I can confirm the authbind + pm2 + config file is not working, resulting in EACCESS.

I.E:
```
authbind --deep pm2 start ./config/pm2-proxy.config.js --env development
```

No matter what alias current user is using (and sourced by bashrc).

Note: authbind configuration if fine.

```
$ authbind node -e "require('http').createServer((request, response) => response.end('Yay\n')).listen(80)" # terminal 1

$ curl localhost:80 # terminal 2
Yay
```

@niftylettuce are you able to use authbind and pm2 start + config file?

**Comment by damianobarbati** [comment]
Wow, very obtrusive... is it going to get lost with pm2 upgrades? Is there any PR to merge this into core? :\

**Comment by niftylettuce** [comment]
No PR exists yet - and as long as you don't overwrite with the `pm2 startup` command it won't get lost with upgrades.  Normal `pm2` upgrades, e.g. `npm install -g pm2@latest` will not affect this.

**Comment by niftylettuce** [comment]
@damianobarbati if you submit a PR that would be incredible, @vmarchaud linked us to the code here <https://github.com/Unitech/pm2/issues/3165#issuecomment-331833056>
